nexus_constructor/json/load_from_json.py

```diff
@@ -99,42 +99,57 @@ def _find_shape_information(children: List[Dict]) -> Union[Dict, None]:
     """
-    for item in children:
-        if item[CommonKeys.NAME] in [SHAPE_GROUP_NAME, PIXEL_SHAPE_GROUP_NAME]:
-            return item
+    try:
+        for item in children:
+            if item[CommonKeys.NAME] in [SHAPE_GROUP_NAME, PIXEL_SHAPE_GROUP_NAME]:
+                return item
+    except KeyError:
+        return None
     return None
 
 
-# Parameter group should be typed with Group but as the inheritance from base class Group
-# by StreamGroup is broken for the attribute children,
-# parameter group cannot be typed here.
-def _add_field_to_group(item: Dict, group):
-    field_type = item[CommonKeys.TYPE]
-    if (
-        field_type == NodeType.STREAM
-    ):  # streams do not have a name field so need to be dealt with differently from other fields.
-        stream = _create_stream(item)
+def _add_field_to_group(item: Dict, group: Group):
+    child: Union[Group, Link]
+    if CommonKeys.TYPE in item:
+        if item[CommonKeys.NAME] == TRANSFORMS_GROUP_NAME:
+            return
+        field_type = item[CommonKeys.TYPE]
+        child_name = item[CommonKeys.NAME]
+        if field_type == NodeType.GROUP:
+            child = _create_group(item, group)
+        elif field_type == NodeType.LINK:
+            child = _create_link(item)
+        else:
+            raise Exception(
+                f"Found unknown field type when loading JSON - {child_name}"
+            )
+        group[child_name] = child
+    elif CommonKeys.MODULE in item:
+        stream: Union[Dataset, Stream]
+        writer_module = item[CommonKeys.MODULE]
+        if writer_module == DATASET:
+            if item[NodeType.CONFIG][CommonKeys.NAME] == CommonAttrs.DEPENDS_ON:
+                return
+            stream = _create_dataset(item, group)
+        else:
+            stream = _create_stream(item)
         group.children.append(
-            stream
+            stream  # type: ignore
         )  # Can't use the `[]` operator because streams do not have a name to use as a key
-    else:
-        child_name = item[CommonKeys.NAME]
-        child: Union[Group, Link, Dataset]
-        if (
-            child_name not in CHILD_EXCLUDELIST
-        ):  # ignore transforms, shape etc as these are handled separately
-            if field_type == NodeType.GROUP:
-                child = _create_group(item, group)
-            elif field_type == NodeType.DATASET:
-                child = _create_dataset(item, group)
-            elif field_type == NodeType.LINK:
-                child = _create_link(item)
-            else:
-                raise Exception(
-                    f"Found unknown field type when loading JSON - {child_name}"
-                )
-            # Add the child to the parent group
-            group[child_name] = child
+
+
+def _find_depends_on_path(items: List[Dict]) -> str:
+    if not isinstance(items, list):
+        raise RuntimeError("Items is not a list.")
```

Comment:
This RuntimeError should probably be more informative. Maybe something like:
"The type of children for {json_object[CommonKeys.NAME]} is not a list."

Reply:
Done.